core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/CustomDimensions.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.internal.statsbeat;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.internal.system.SystemInformation;
+import com.microsoft.applicationinsights.internal.util.PropertyHelper;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.*;
+
+public final class CustomDimensions {
+
+    private static CustomDimensions instance;
+    private String resourceProvider;
+    private String operatingSystem;
+    private String customerIkey;
+    private String version;
+    private String runtimeVersion;
+    private ConcurrentMap<String, String> properties;
+    private static final Object lock = new Object();
+    
+    public static CustomDimensions getInstance() {
+        synchronized (lock) {
+            if (instance == null) {
+                instance = new CustomDimensions();
+            }
+        }
+        return instance;
+    }
+    
+    private CustomDimensions() {
+        initialize();
+    }
+
+    private void initialize() {
```

Comment:
I don't think separate method adds value in this case, since outer method is only single line

also, then can make fields above `final`?

```suggestion
```